Log debug message if import PythonShell and PythonEditor fail in api.py

kitchoi opened the pull request:
Follow up from https://github.com/enthought/pyface/pull/796
In particular, this: https://github.com/enthought/pyface/pull/796#issuecomment-725538154

If PythonEditor and PythonShell cannot be imported because pygments is not available, it might help if we log a debug message so that downstream projects have more information when the import fails.

The alternative would be to emit a warning, but for developers who don't care about these two classes, this could be too noisy.

corranwebster commented on pyface/util/_optional_dependencies.py:
```diff
@@ -0,0 +1,40 @@
+import logging
+
+
+_PYFACE_LOGGER = logging.getLogger("pyface")
```
This makes me a bit twitchy, having the logger being used hard-coded.  Perhaps better would be to pass in the logger instance to `optional_import` so it can use the logger of the module where `optional_import` is called from.

corranwebster replied:
It would also make testing easier

corranwebster commented on pyface/api.py:
```diff
@@ -28,13 +28,16 @@
+from .util._optional_dependencies import optional_import as _optional_import
+
+# Excuse pygments dependency (for Qt), otherwise re-raise
+with _optional_import("pygments", msg="PythonEditor not available"):
```
The log message doesn't mention that the reason it isn't available is because Pygments isn't installed - that would be useful to know.

corranwebster commented on pyface/util/_optional_dependencies.py:
```diff
@@ -0,0 +1,40 @@
+        yield
+    except ImportError as exception:
+        if exception.name == dependency_name:
+            _PYFACE_LOGGER.debug(msg, exc_info=True)
```
Maybe `info` rather than `debug`?

kitchoi replied:
I still prefer debug because this is intended for debugging purposes. But that is not a strong preference.

kitchoi replied:
I think typically info message does not contain exception traceback info (the `exc_info=True`) bit. Here I think we want to show the traceback which will include the ImportError on pygments.
I could do both, I suppose?

Like this?
https://github.com/enthought/pyface/blob/7e32dbfd55de290ecde1da3ef792ff5be5da40c1/pyface/base_toolkit.py#L240-L241

kitchoi commented:
Thank you @corranwebster 
I have addressed the comments, apart from the logging level: I am not entirely sure which of `info` or `debug` we should use or if we should log both.

corranwebster reviewed (approved):
LGTM
